upgrade/upgrade-scripts/rootfs-cleanup

```diff
@@ -0,0 +1,161 @@
+    ]).decode("UTF-8").splitlines()
+
+    #
+    # We remove the first entry in the list, as that'll refer to the "rpool/ROOT"
+    # dataset, which is not versioned and also not eligible for deletion.
+    #
+    filesystems = filesystems[1:]
+
+    current = subprocess.check_output(["get-appliance-version"
+                                      ]).decode("UTF-8").strip()
+
+    #
+    # Our policy is to keep two delphix versions at any given time; the
+    # currently running version, and the version used prior to the
+    # current one.
+    #
+    # By sorting the list of all filesystems and snapshots on the sytem,
+    # we can easily determine the highest two versions on the system,
+    # and skip those versions (to maintain the policy described above);
+    # this is based on the assumption that the versions of an appliance
+    # will always increase (or remain constant, in the case of rollback)
+    # after an upgrade (i.e. we never upgrade to a lower version).
+    #
+    for rootfs in sorted(filesystems + snapshots, key=rootfscmp)[:-2]:
+        #
+        # In the event of a rollback, we want to be careful to not
+        # delete the currently running version, as well as any versions
+        # that may have failed to upgrade resulting in the rollback
+        # event to begin with. Thus, to account for the exceptional case
+        # of rollback, we simply skip any version that's equal to or
+        # greater than what's currently running.
+        #
+        if dpkgcmp(version(rootfs), "ge", current):
+            break
+
+        #
+        # If a rollback event has occured, it's possible that we'll be
+        # unable to delete a snapshot or filesystem; e.g. if it's still
+        # being used as the "origin" of another filesystem. Rather than
+        # adding more complicated logic to handle this situation, we
+        # simply allow the commands to throw an error.
+        #
+        # If we fail to remove a snapshot or filesystem, this shouldn't
+        # negatively impact the appliance, apart from potentially
+        # consuming some more storage than strictly necessary.
+        #
+        # Further, as the appliance continues to be upgraded to new
+        # versions over time, these error conditions will subside; it
+        # just might take a couple upgrades to allow the previously
+        # "pinned" version to be removed and cleaned up.
+        #
+        # Lastly, it's possible for us to delete an entire rootfs
+        # container early in the iteration, and then attempt to delete
+        # some of that container's snapshots later in the iteration.
+        # Since deleting the container will either delete (or promote)
+        # all of that container's snapshots, the attempt (later) to
+        # delete the snapshots individually will silently fail.
+        #
+        if rootfs in filesystems:
+            subprocess.run([
+                "{}/rootfs-container".format(IMAGE_PATH), "delete",
```

Comment:
Would this command be logged somewhere easily accessible?
It might be useful to add a quick print that mentions that a container is being deleted.

Reply:
it'll be captured in syslog.